EasyPost/Hooks.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+
+namespace EasyPost;
+
+public class Hooks
+{
+    /// <summary>
+    ///     An <see cref="EventHandler{OnRequestExecutingEventArgs}"/> to view an HTTP request by the client prior to being sent.
+    ///     Editing the <see cref="HttpRequestMessage"/> in this callback does not impact the <see cref="HttpRequestMessage"/> being executed.
+    /// </summary>
+    public EventHandler<OnRequestExecutingEventArgs>? OnRequestExecuting { get; set; }
+
+    /// <summary>
+    ///     An <see cref="EventHandler{OnRequestResponseReceivedEventArgs}"/> to view an HTTP response received by the client.
+    /// </summary>
+    public EventHandler<OnRequestResponseReceivedEventArgs>? OnRequestResponseReceived { get; set; }
+}
+
+public class OnRequestExecutingEventArgs : EventArgs
+{
+    /// <summary>
+    ///     The <see cref="HttpRequestMessage"/> about to be executed by the HTTP request.
+    /// </summary>
+    private HttpRequestMessage Request { get; } // Not publicly-exposed, used for reference only.
+
+    /// <summary>
+    ///     The <see cref="HttpMethod"/> of the HTTP request.
+    /// </summary>
+    public HttpMethod Method => Request.Method;
+
+    /// <summary>
+    ///     The <see cref="Uri"/> of the HTTP request.
+    /// </summary>
+    public Uri? Uri => Request.RequestUri;
+
+    /// <summary>
+    ///     The <see cref="HttpContent"/> of the HTTP request.
+    /// </summary>
+    public HttpContent? RequestBody => Request.Content;
+
+    /// <summary>
+    ///     The <see cref="HttpHeaders"/> of the HTTP request.
+    /// </summary>
+    public HttpRequestHeaders Headers => Request.Headers;
+
+    /// <summary>
+    ///     The timestamp of the HTTP request.
+    /// </summary>
+    public int RequestTimestamp { get; }
+
+    /// <summary>
+    ///     A unique identifier for the HTTP request-response pair.
+    /// </summary>
+    public Guid Id { get; }
```

Comment:
This need to be renamed to `RequestUuid`

Reply:
Pushing back against this one, since it's not really a request Uuid, it's a GUID of a request-response pair.